website/src/components/docs/library/api/MethodSignature.tsx

```diff
@@ -0,0 +1,128 @@
+import styled from "@emotion/styled";
+import { useTypescriptSnippets } from "@site/src/components/TypeScriptToggle";
+import React, { FC } from "react";
+import { Parameter, Signature, Type } from "./getLibraryMethods";
+
+interface Props {
+  methodName: string;
+  signature: Signature;
+}
+
+const MethodSignature: FC<Props> = ({ methodName, signature }) => {
+  const showTypeScriptSnippets = useTypescriptSnippets();
+
+  const { shortText, parameters, returns, returnType } = signature;
+
+  const getParameter = (parameter: Parameter) => {
+    if (showTypeScriptSnippets) {
+      return parameter.name + ": " + formatType(parameter.type);
+    }
+
+    return parameter.name;
+  };
+
+  let title = methodName + "(" + parameters.map(getParameter).join(", ") + ")";
+
+  if (showTypeScriptSnippets) {
+    title += ": " + formatType(returnType);
+  }
+
+  return (
+    <div>
+      <code>{title}</code>
+      <p>{shortText}</p>
+
+      <h4>Parameters</h4>
+      <ul>
+        {parameters.map(parameter => (
+          <li key={parameter.name}>
+            <p>
+              <code>
+                {parameter.name}
+                {showTypeScriptSnippets && ": " + formatType(parameter.type)}
+              </code>
+              {!!parameter.shortText &&
+                parameter.shortText.length > 0 &&
+                " - " + parameter.shortText}
+            </p>
+          </li>
+        ))}
+      </ul>
+
+      <h4>Returns</h4>
+      <Returns>
+        <code>{formatType(returnType)}</code>
+        {!!returns && returns.length > 0 && " - " + returns}
+      </Returns>
+
+      <hr />
+    </div>
+  );
+};
+export default MethodSignature;
+
+const formatType = (type: Type) => {
```

Comment:
I think this is the same function/logic as the one in the LibraryMethod component? Do you think it's worth extracting into a common function? I know there are sometimes justifications for not sharing logic but just want to check this wasn't like a copy/paste artefact.